src/bowlingcard.js

```diff
@@ -0,0 +1,84 @@
+'use strict';
+
+function Bowlingcard () {
+  this.score = 0;
+  this.roll = 0;
+  this.rollScores = [];
+  this.frameArray = [];
+}
+
+Bowlingcard.prototype.enterScore = function(number) {
+  //can this be a while loop instead?
+  this.endGame();
+  //this test is failing but seems to work, check test
+  if (number > 10) {
+    return this.inputError();
+  } else {
+    //lines 17 - 18 can be its own method of like the rest below it
+    this.rollScores.push(number);
+    this.score += number;
+    this.roll++;
+    this.frameArrayAdd();
+    this.strikeScore();
+    this.spareScore();
+    this.isStrike();
+  }
+}
+
+Bowlingcard.prototype.inputError = function() {
+    return "what?"
+}
+
+Bowlingcard.prototype.isStrike = function() {
+  if (this.roll % 2 != 0 && this.rollScores[this.roll - 1] === 10) {  
+    this.rollScores.push(0);
+    this.roll++; 
+  }
+}
+
+Bowlingcard.prototype.strikeScore = function() {
+  const strike1FrameBeforeRoll1 = this.roll % 2 != 0 && this.rollScores[this.roll - 3] === 10;
+  const strike1FrameBeforeRoll2 = this.roll % 2 === 0 && this.rollScores[this.roll - 4] > 9;
+  const strike2FramesBefore = strike1FrameBeforeRoll1 && this.rollScores[this.roll - 5] > 9;
+  if (strike2FramesBefore) {
+    //can this be refactored into a finction?
+    this.rollScores[this.roll - 5] = this.rollScores[this.roll - 1] + this.rollScores[this.roll - 5];
+    this.rollScores[this.roll - 3] = this.rollScores[this.roll - 1] + this.rollScores[this.roll - 3]; 
+  } else if (strike1FrameBefore) {
+  this.rollScores[this.roll - 3] = this.rollScores[this.roll - 1] + this.rollScores[this.roll - 3];
+  } else if (strike1FrameBeforeRoll2) {
+    this.rollScores[this.roll - 4] = this.rollScores[this.roll - 1] + this.rollScores[this.roll - 4]; 
```

Comment:
could `this.rollScores[this.roll - x] ` be extracted into a variable, or even a method?